crates/pixi/tests/integration_rust/update_tests.rs

```diff
@@ -156,54 +159,179 @@ async fn test_update_single_package() {
+            "tqdm",
+        )
+        .unwrap();
+
+    let tqmd_version = tqmd_package.as_pypi().unwrap().0.version.to_string();
+
+    let mut lock_file_str = lock.render_to_string().unwrap();
+
+    // git url should have a fragment
+    let fragment = tqmd_package
+        .as_pypi()
+        .unwrap()
+        .0
+        .location
+        .as_url()
+        .unwrap()
+        .fragment()
+        .expect("expected git url to have a fragment");
+
+    // and modify this fragment to simulate an older commit
+    let older_commit = "a2d5f1c9d1cbdbcf56f52dc4365ea4124e3e33f7";
+    lock_file_str = lock_file_str.replace(fragment, older_commit);
+
+    lock_file_str = lock_file_str.replace(&tqmd_version, "4.67.1.dev5+ga2d5f1c9d");
+
+    let lockfile = LockFile::from_str(&lock_file_str).unwrap();
+
+    lockfile.to_path(&workspace.lock_file_path()).unwrap();
+
+    // now run the update command to update conda packages
+    // which will invalidate also pypi packages
+    pixi.update().with_package("python").await.unwrap();
+
+    // Get the re-locked lock-file
+    let lock = pixi.lock_file().await.unwrap();
+
+    let url_or_path = lock
+        .get_pypi_package_url(
+            consts::DEFAULT_ENVIRONMENT_NAME,
+            Platform::current(),
+            "tqdm",
+        )
+        .unwrap();
+    let new_fragment = url_or_path
+        .as_url()
+        .unwrap()
+        .fragment()
+        .expect("expected git url to have a fragment");
+    assert_eq!(
+        older_commit, new_fragment,
+        "expected git pypi package to not be updated when updating conda packages"
+    );
+}
+
+#[tokio::test]
+async fn test_update_conda_package_doesnt_update_git_pypi_pinned() {
+    setup_tracing();
+
+    let pixi = PixiControl::new().unwrap();
+
+    // Create a new project using our package database.
+    pixi.init().await.unwrap();
+
+    // Add a dependency on `python`
+    pixi.add("python").await.unwrap();
+
+    // Add a `pinned` git pypi dependency on `tqdm`
+    // this should not trigger an update
+    pixi.add_pypi(
+        "tqdm @ git+https://github.com/tqdm/tqdm.git@cac7150d7c8a650c7e76004cd7f8643990932c7f",
+    )
+    .await
+    .unwrap();
+
+    // Get the created lock-file
+    let lock = pixi.lock_file().await.unwrap();
+
+    // previous lockfile
+    let previous_lockfile_str = lock.render_to_string().unwrap();
+
+    // now run the update command to update conda packages
+    // which should not trigger any update for the pinned pypi package
+    pixi.update().with_package("python").await.unwrap();
+
+    // Get the re-locked lock-file
+    let lock = pixi.lock_file().await.unwrap();
+
+    let new_lockfile_str = lock.render_to_string().unwrap();
+
+    assert_eq!(
+        previous_lockfile_str, new_lockfile_str,
+        "expected git pypi package to not be updated when updating conda packages"
+    );
+}
+
+#[tokio::test]
+async fn test_update_conda_package_update_git_pypi_when_requested() {
```

Comment:
I guess the `update_conda_package` does not really hold as we only update the PyPI package?